refactor(friend): 친구 페이지 UI 개선 및 API 연결

hakilee opened the pull request:
## 🔗 반영 브랜치

- closed #113
- closed #139
- `refactor/friend → dev`

## 📝 작업 내용

- 친구 페이지에 구현되어 있는 일부 UI를 개선했습니다.
- 다이얼로그 내부에 API를 연결했습니다.
- `status` 소켓을 전역 상태로 관리할 수 있게 변경했습니다.

### 📸 스크린샷
<img width="798" alt="Screenshot 2025-05-18 at 19 22 03" src="https://github.com/user-attachments/assets/fed827c9-6fd6-40f1-b665-c140a5273fb2" />
<img width="798" alt="Screenshot 2025-05-18 at 19 24 15" src="https://github.com/user-attachments/assets/6ea48a00-a9eb-4d69-9881-9a30c3ee543b" />

hakilee committed:
refactor: rename FriendChatRoomPage to ChatRoomPage

hakilee committed:
Merge remote-tracking branch 'origin/dev' into refactor/friend

hakilee committed:
feat: implement friend request dialog

This commit introduces a friend request dialog to
handle friend requests.

The changes include:
- Created a new component `FriendRequestDialog` for
  displaying friend requests.
- Added styles for the dialog content.
- Replaced the `RequestListDialog` with the new
  `FriendRequestDialog` in the `FriendPage`.
- Adjusted the margin of the alarm icon in the
  `FriendPage` styles.

This change improves the user experience by providing
a dedicated dialog for managing friend requests.

hakilee committed:
feat: implement user search functionality

this commit introduces the functionality to search for users based on nickname and friend request status.

the following changes were made:

- added `userssearchpayload` type to `src/api/types/user.ts` to define the structure of the search payload.
- added `userssearch` query key to `src/api/querykey.ts` to define the query for searching users.
- created `src/api/queries/useuserssearch.ts` to provide react query hooks for the users search functionality.
- exported `useuserssearch` and `usesuspenseuserssearch` from `src/api/queries/index.ts`.
- updated dialog title element on friend request dialog

this feature allows users to easily find other users and manage friend requests.

there are no breaking changes introduced in this commit.

hakilee committed:
feat: implement friend search dialog and requests

hakilee committed:
feat: integrate friend request and search functionality

this commit introduces the friend request feature. it includes:

- removal of the previous add friend component and related styles.
- integration of a friend search dialog for sending friend requests.
- creation of a friend request dialog to display pending requests.
- implementation of api queries and mutations for fetching friend requests and creating new requests.
- styling updates for the friend request dialog and search dialog.

there are no breaking changes. the new components and api calls enhance the friend management capabilities.

hakilee committed:
chore: remove request list dialog component

this commit removes the `requestlistdialog` component and its
associated styles.

this component was deemed unnecessary for the current
implementation and was removed to simplify the codebase.

there are no breaking changes.

hakilee committed:
style: improve friend request and search dialog styles

This commit improves the styling of the friend request and search dialogs for better user experience.

The following changes were made:

- Added marginTop to the list of friend requests for better spacing.
- Added a buttonGroup style for friend request actions.
- Added user-select: 'none' to buttons in both dialogs to prevent text selection on click.
- Replaced the buttons "Accept" and "Decline" in the friend request dialog by icon buttons
- Changed <p> tag by <span> tag in friend search dialog, to fix a nesting problem

These changes enhance the visual appeal and usability of the friend-related dialogs.

hakilee committed:
feat: implement accept/reject friend request functionality

this commit introduces the functionality to accept or
reject friend requests. it includes the creation of
new mutation hooks `useAcceptFriendsRequests` and
`useRejectFriendsRequests` to handle the api calls.
the friend request dialog now includes accept and
reject buttons, which trigger the respective mutation.
on success, a toast message is displayed. on error,
an error message is logged and a toast message is
displayed. the button group spacing in the dialog has
been increased for better usability.

hakilee committed:
fix: use patch instead of post for friend request mutations

This commit fixes a bug where friend requests were not being properly accepted or rejected due to using the incorrect HTTP method. The API endpoints for accepting and rejecting friend requests require a PATCH request, but the code was using POST.

This change updates the `useAcceptFriendsRequests` and `useRejectFriendsRequests` mutations to use the `patch` method instead of `post` when making the API calls. This ensures that the requests are correctly processed by the server and that friend requests are accepted and rejected as expected.

There are no breaking changes introduced by this commit.

hakilee committed:
feat: invalidate friends requests on accept/reject

This change invalidates the `friendsRequests` query after accepting or rejecting a friend request. This ensures that the UI immediately reflects the change in the number of pending friend requests without requiring a manual refresh.

The `friendsRequests` query is now invalidated within the `onSuccess` callback of the `useAcceptFriendsRequests` and `useRejectFriendsRequests` mutations.

This change does not introduce any breaking changes.

HJ0311 commented on src/pages/FriendPage/components/friend-request-dialog/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  DialogPortal,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/system';
```
친구 페이지랑 매칭 페이지에서 쓰던 팝업 관련 다이얼로그 공통  components 폴더로 전부 뺐습니다. #156 PR에 적용되었는데요. 이 부분은 머지 이후 제가 다시 한 번 수정해야 될 것 같습니다!

hakilee replied:
다이얼로그는 공통으로 관리하지 않는 것이 더 좋을 것 같아서 분리했습니다!

HJ0311 reviewed (dismissed):
수고하셨습니다!
이전에 제가 작업하던 친구 리스트에서 파일 몇 개가 추가 된 것 같아 보이는데요.
피알에 무엇이 추가된 건지에 대한 내용이 없어서 정확하게 파악이 안 되어서 질문드립니다.
친구 리스트에서 팝업을 띄우던 버튼과 관련된 다이얼로그들이 새롭게 추가된 것 같은데 맞나요?
원래 친구 페이지에 friend search dialog가 없었던 것 같은데 추가 된 것 같아서요.

hakilee committed:
Merge remote-tracking branch 'origin/dev' into refactor/friend

hakilee committed:
feat: redesign friend list and integrate status updates

this commit restructures the friend list component to
enhance its visual appearance and integrate real-time
status updates. the previous friend item component and
general alarm components have been removed.

the changes include:

- replacement of the friend item with a profile card for
  a cleaner design.
- integration of user status using a jotai atom and web
  socket for real-time updates.
- addition of a divider component for visual separation
  between friends.
- implementation of a search functionality with debouncing
  for efficient filtering.

these changes improve the user experience by providing a
more visually appealing and informative friend list. the
integration of real-time status updates allows users to
see the current status of their friends, facilitating
better communication and coordination. there are no
breaking changes introduced by this commit.

hakilee commented on src/api/mutations/useAcceptFriendsRequests.ts:
```diff
@@ -0,0 +1,28 @@
+  return useMutation({
+    mutationFn: patchAcceptFriendsRequests,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries(queryKeys.friendsRequests());
+      await queryClient.invalidateQueries({ ...queryKeys.friendsMe(), refetchType: 'all' });
+      await queryClient.invalidateQueries({
+        queryKey: [queryKeys.usersSearch({ nickname: '', exceptMe: true, status: 'NONE' })._def],
+        refetchType: 'all',
+      });
```
중복을 허용하는 것이 더 이해하기 쉽습니다.

HJ0311 reviewed (approved):
수고하셨습니다! 변경 사항 확인했습니다
